dulwich/porcelain.py

```diff
@@ -97,6 +98,14 @@
 GitStatus = namedtuple('GitStatus', 'staged unstaged untracked')
 
 
+if sys.version_info[0] < 3:
+    STDOUT = sys.stdout
+    STDERR = sys.stderr
+else:
+    STDOUT = codecs.getwriter('utf-8')(sys.stdout.buffer)
```

Comment:
The system encoding is not necessarily utf-8. Is there any reason why just sys.stdout.buffer wouldn't work?